_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,420 @@
+---
+layout: post
+title:  "Reassigning partitions in Apache Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+As an Apache Kafka user, we sometimes have to scale up/down the number of Kafka brokers in our cluster depending on the use case. Addition of extra brokers can be an advantage to handle massive load, and we can use Cruise Control for general rebalancing in Strimzi since it allows us to automate the balancing of load across the cluster but what if we are scaling down the clusters. Let us understand this with the help of an example, suppose there are certain number of brokers in a cluster and now we want to remove a broker from the cluster. We need to make sure that the broker which is going to be removed should not have any assigned partitions. Strimzi's integration of Cruise Control currently doesn't support scaling down the cluster. This requires a tool that can assign the partitions from the broker to be removed, to the remaining brokers. The most convenient tool for this job is the Kafka reassignment partition tool: `kafka-reassign-partitions.sh`.
```

Comment:
General advice I make everytime to people starting to write blog posts or proposals: it's much better having one sentence per line, because it helps reviewers to make specific suggestion.
For example, in this case if I want to make a suggestion on the first line but then on the fifth line, I can make just one comment here because how GitHub works. Having the sentences on different lines helps on comment as well.
In this case I am lucky :-) ... my comment is about the last line ... I would avoid to specify the name of the shell script but just mentioning "Kafka reassignment partition tool".